test/x11.c

```diff
@@ -84,3 +85,7 @@ main(void)
 
     return exit_code;
 }
+
+int main(void) {
+    xvfb_wrapper(&test);
+}
```

Comment:
side-note: you could wrap this more generically by plonking the actual tests into a special ELF section and then have the *wrapper* loop through the sections to execute each test individually. This way the tests stay pristine with no knowledge on how they're invoked.

And you could easily duplicate the wrapper with a non-Xvfb version without having to touch the tests - just recompile with a different `wrapper.c`.

libei does this with the [util-munit.h]([https://gitlab.freedesktop.org/libinput/libei/-/blob/main/src/util-munit.h]) and [util-munit.c](https://gitlab.freedesktop.org/libinput/libei/-/blob/main/src/util-munit.c) so that the actual invocation of the unit test (in the current binary) is a simple call to `run_all`, see [unit-tests.c](https://gitlab.freedesktop.org/libinput/libei/-/blob/main/test/unit-tests.c)
Note that libei uses munit as test suite but those bits are quite trivially removed.

Reply:
I gave it a try. I am not sure this is much better.

Reply:
it's not going to look like much until you have to add other tests to it - that's when it'll pay off because there's no manually maintained test registry anymore.